lib/bigtable/family.js

```diff
@@ -0,0 +1,311 @@
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the family exists or not.
+     *
+     * @example
+     * family.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get a column family if it exists.
+     *
+     * You may optionally use this to "get or create" an object by providing an
+     * object with `autoCreate` set to `true`. Any extra configuration that is
+     * normally required for the `create` method must be contained within this
+     * object as well.
+     *
+     * @param {options=} options - Configuration object.
+     * @param {boolean} options.autoCreate - Automatically create the object if
+     *     it does not exist. Default: `false`
+     *
+     * @example
+     * family.get(function(err, family, apiResponse) {
+     *   // `family.metadata` has been populated.
+     * });
+     */
+    get: true
+  };
+
+  var config = {
+    parent: table,
+    id: id,
+    methods: methods,
+    createMethod: function(_, options, callback) {
+      table.createFamily(name, options, callback);
+    }
+  };
+
+  GrpcServiceObject.call(this, config);
+}
+
+nodeutil.inherits(Family, GrpcServiceObject);
+
+/**
+ * Format the Column Family name into the expected proto format.
+ *
+ * @private
+ *
+ * @param {string} tableName - The full formatted table name.
+ * @param {string} name - The column family name.
+ * @return {string}
+ *
+ * @example
+ * Family.formatName_(
+ *   'projects/p/zones/z/clusters/c/tables/t',
+ *   'my-family'
+ * );
+ * // 'projects/p/zones/z/clusters/c/tables/t/columnFamilies/my-family'
+ */
+Family.formatName_ = function(tableName, name) {
+  if (name.indexOf('/') > -1) {
+    return name;
+  }
+
+  return tableName + '/columnFamilies/' + name;
+};
+
+/**
+ * Formats Garbage Collection rule into proto format.
+ *
+ * @private
+ *
+ * @param {object} ruleObj - The rule object.
+ * @return {object}
+ *
+ * @example
+ * Family.formatRule({
+ *   age: {
+ *     seconds: 10000,
+ *     nanos: 10000
+ *   },
+ *   versions: 2,
+ *   union: true
+ * });
+ * // {
+ * //   union: {
+ * //     rules: [
+ * //       {
+ * //         maxAge: {
+ * //           seconds: 10000,
+ * //           nanos: 10000
+ * //         }
+ * //       }, {
+ * //         maxNumVersions: 2
+ * //       }
+ * //     ]
+ * //   }
+ * // }
+ */
+Family.formatRule_ = function(ruleObj) {
+  var rules = [];
+
+  if (ruleObj.age) {
+    rules.push({
+      maxAge: ruleObj.age
+    });
+  }
+
+  if (ruleObj.versions) {
+    rules.push({
+      maxNumVersions: ruleObj.versions
+    });
+  }
+
+  if (!ruleObj.intersection && !ruleObj.union) {
+    return rules[0];
+  }
+
+  var rule = {};
+  var ruleType = ruleObj.union ? 'union' : 'intersection';
+
+  rule[ruleType] = {
+    rules: rules
+  };
+
+  return rule;
+};
+
+/**
+ * Get the family metadata.
```

Comment:
This reads a little robotic to me. Can we liven it up, like "Get the column family's metadata." or "Get the current metadata of the column family."?